client_integration_test.go

```diff
@@ -853,6 +853,15 @@ func TestClientChmod(t *testing.T) {
 	} else if stat.Mode()&os.ModePerm != 0531 {
 		t.Fatalf("invalid perm %o\n", stat.Mode())
 	}
+
+	sf, err := sftp.Open(f.Name())
+	require.NoError(t, err)
+	defer sf.Close()
```

Comment:
Let’s move the `sf.Chmod()` to before this `defer sf.Close()` and then remove the `defer`.

If we add further tests to the end of this function, the `defer` will end up waiting all the way until after that other code, so scoping it more correctly here is probably a good idea.

Reply:
Done.